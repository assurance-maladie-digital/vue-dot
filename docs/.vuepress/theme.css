html, body {
	font-size: 100% !important;
}

.v-ripple__container,
.v-ripple__container * {
	max-width: none;
}

.v-input--selection-controls__ripple {
	max-width: none;
}

.application--wrap {
	width: 100%;
}

.theme--light.v-input--is-disabled .v-label,
.theme--light.v-icon.v-icon--disabled,
.theme--light.v-list .v-list--disabled {
	color: #767676 !important;
}

.theme--light.v-btn.v-btn--disabled .v-icon {
	color: #949494 !important;
}

.v-tabs__bar .v-tabs__item--disabled {
	color: #333 !important;
}

.theme--light.v-btn.v-btn--disabled {
	color: #646464 !important;
}

.content h1 {
	font-size: 2.2rem;
	margin-bottom: 25px;
}

.content h2 {
	font-size: 1.65rem;
	margin-bottom: 16px;
	margin-top: 35px;
}

.content h2 + h3 {
	margin-top: 0;
}

.content h3 {
	font-size: 1.35rem;
	margin-bottom: 10px;
	margin-top: 35px;
}

.content h3 + h4 {
	margin-top: 0;
}

.content h4 {
	font-size: 1.15rem;
	margin-bottom: 5px;
	margin-top: 15px;
}

pre[class*='language-'] {
	align-items: center;
	box-shadow: none;
	display: flex;
	border-radius: 2px;
	position: relative;
	overflow-x: auto;
	overflow-y: hidden;
	background: #2d2d2d;
	margin: 0;
	margin-bottom: 5px;
}

code[class*='language-'], code {
	position: relative;
	box-shadow: none;
	overflow-x: auto;
	overflow-y: hidden;
	word-break: break-word;
	flex-wrap: wrap;
	align-items: center;
	vertical-align: middle;
	white-space: pre-wrap;
	margin: 0;
	font-family: 'Inconsolata', monospace;
	font-weight: 300;
	font-size: .9rem;
	line-height: 1.55;
	padding: 0 5px;
}

code {
	background: rgba(240, 240, 240, .8);
	color: rgba(0, 0, 0, .7);
}

code::after,
code::before {
	content: none;
}

pre code {
	background: transparent;
	padding: 0;
	color: #ccc;
}

.header-anchor {
	color: #0c4887;
	transition: opacity .15s !important;
	text-decoration: none;
	font-size: .85em;
	float: left;
	margin-left: -.8em;
	margin-top: .125em;
	padding-right: .23em;
	opacity: 0;
	position: absolute;
}

[id]:hover > .header-anchor,
.header-anchor:focus {
	opacity: 1;
}

.icon.outbound {
	color: inherit;
}

.custom-block {
	padding: .1rem 1.5rem;
	border-left-width: .5rem;
	border-left-style: solid;
	margin: 1rem 0;
	background-color: #f3f5f7 !important;
	border-color: #42b983;
}

.custom-block.warning {
	background-color: rgba(255, 229, 100, .3) !important;
	border-color: #e7c000 !important;
}

.custom-block .custom-block-title {
	font-weight: 600;
	margin-top: 10px;
	margin-bottom: 0;
}

.custom-block.warning .custom-block-title {
	color: #776300;
}

.custom-block.danger {
	background-color: #ffe6e6 !important;
	border-color: #c00 !important;
}

.custom-block.danger .custom-block-title {
	color: #4d0000;
}
