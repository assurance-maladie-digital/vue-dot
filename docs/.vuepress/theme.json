{
	"name": "digital",
	"version": "0.0.1",
	"config": {
		"styles": {
			"html, body": {
				"font-size": "100% !important"
			},
			".v-ripple__container, .v-ripple__container *": {
				"max-width": "none"
			},
			".v-input--selection-controls__ripple": {
				"max-width": "none"
			},
			".application--wrap": {
				"width": "100%"
			},
			".theme--light.v-input--is-disabled .v-label, .theme--light.v-icon.v-icon--disabled, .theme--light.v-list .v-list--disabled": {
				"color": "#767676 !important"
			},
			".theme--light.v-btn.v-btn--disabled .v-icon": {
				"color": "#949494 !important"
			},
			".v-tabs__bar .v-tabs__item--disabled": {
				"color": "#333 !important"
			},
			".theme--light.v-btn.v-btn--disabled": {
				"color": "#646464 !important"
			},
			".content h1": {
				"font-size": "2.2rem",
				"margin-bottom": "25px"
			},
			".content h2": {
				"font-size": "1.65rem",
				"margin-bottom": "16px",
				"margin-top": "35px"
			},
			".content h2 + h3": {
				"margin-top": "0"
			},
			".content h3": {
				"font-size": "1.35rem",
				"margin-bottom": "10px",
				"margin-top": "35px"
			},
			".content h3 + h4": {
				"margin-top": "0"
			},
			".content h4": {
				"font-size": "1.15rem",
				"margin-bottom": "5px",
				"margin-top": "15px"
			},
			"pre[class*='language-']": {
				"align-items": "center",
				"box-shadow": "none",
				"display": "flex",
				"border-radius": "2px",
				"position": "relative",
				"overflow-x": "auto",
				"overflow-y": "hidden",
				"background": "#2d2d2d",
				"margin": "0",
				"margin-bottom": "5px"
			},
			"code[class*='language-'], code": {
				"position": "relative",
				"box-shadow": "none",
				"overflow-x": "auto",
				"overflow-y": "hidden",
				"word-break": "break-word",
				"flex-wrap": "wrap",
				"align-items": "center",
				"vertical-align": "middle",
				"white-space": "pre-wrap",
				"margin": "0",
				"font-family": "'Inconsolata', monospace",
				"font-weight": "300",
				"font-size": ".9rem",
				"line-height": "1.55",
				"padding": "0 5px"
			},
			"code": {
				"background": "rgba(240, 240, 240, .8)",
				"color": "rgba(0, 0, 0, .7)"
			},
			"code::after, code::before": {
				"content": "none"
			},
			"pre code": {
				"background": "transparent",
				"padding": "0",
				"color": "#ccc"
			},
			".header-anchor": {
				"color": "#0c4887",
				"transition": "opacity .15s",
				"text-decoration": "none",
				"font-size": ".85em",
				"float": "left",
				"margin-left": "-.8em",
				"margin-top": ".125em",
				"padding-right": ".23em",
				"opacity": "0",
				"position": "absolute"
			},
			"[id]:hover > .header-anchor, .header-anchor:focus": {
				"opacity": "1"
			},
			".icon.outbound": {
				"color": "inherit"
			},
			".custom-block": {
				"padding": ".1rem 1.5rem",
				"border-left-width": ".5rem",
				"border-left-style": "solid",
				"margin": "1rem 0",
				"background-color": "#f3f5f7 !important",
				"border-color": "#42b983"
			},
			".custom-block.warning": {
				"background-color": "rgba(255, 229, 100, .3) !important",
				"border-color": "#e7c000 !important"
			},
			".custom-block .custom-block-title": {
				"font-weight": "600",
				"margin-top": "10px",
				"margin-bottom": "0"
			},
			".custom-block.warning .custom-block-title": {
				"color": "#776300"
			},
			".custom-block.danger": {
				"background-color": "#ffe6e6 !important",
				"border-color": "#c00 !important"
			},
			".custom-block.danger .custom-block-title": {
				"color": "#4d0000"
			}
		},
		"colors": {
			"primary": "#0c4887",
			"secondary": "#01579b",
			"accent": "#0079bc",
			"error": "#e54d42",
			"info": "#0077c5",
			"success": "#2d9437",
			"warning": "#977000"
		},
		"icons": [
			{
				"name": "vuejs",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 400 400\"><path fill=\"#4dba87\" d=\"M237.42 86.66L207.19 139l-30.22-52.35H76.3l130.9 226.69L338.07 86.66z\"/><path fill=\"#435466\" d=\"M237.42 86.66L207.19 139l-30.22-52.35h-48.3l78.52 136 78.53-136z\"/></svg>"
			},
			{
				"name": "vue-dot-dark",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 400 400\"><path fill=\"#fff\" d=\"M184.55 78.53H61.8l145.4 251.84L352.6 78.53H229.84l-22.64 39.21-22.64-39.21z\"/><path fill=\"#1867c0\" d=\"M237.42 86.66L207.19 139l-30.22-52.35H76.3L207.2 313.34 338.07 86.66z\"/><path fill=\"#1697f6\" d=\"M237.42 86.66L207.19 139l-30.22-52.35h-48.3l78.52 136 78.53-136z\"/></svg>"
			},
			{
				"name": "github",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 24 24\" fill=\"currentColor\"><path d=\"M22.39 6.27A11.95 11.95 0 0 0 12 .29a11.95 11.95 0 0 0-12 12c0 2.62.76 4.97 2.29 7.06a11.75 11.75 0 0 0 5.91 4.33c.28.06.5.02.63-.1.13-.13.2-.29.2-.47a359.71 359.71 0 0 0-.01-2.24l-.36.06a6.6 6.6 0 0 1-1.95-.05 2.43 2.43 0 0 1-1.05-.47 1.98 1.98 0 0 1-.7-.96l-.15-.36a3.9 3.9 0 0 0-.49-.8c-.22-.29-.45-.49-.68-.59l-.1-.08a1.15 1.15 0 0 1-.35-.4c-.03-.08 0-.14.08-.18.08-.05.23-.07.45-.07l.31.04c.2.04.47.17.77.38.31.2.56.48.76.8.24.44.53.76.87.99.34.22.68.33 1.02.33.35 0 .64-.02.9-.08.24-.05.48-.13.7-.23.09-.7.35-1.24.76-1.61a10.7 10.7 0 0 1-1.6-.28 6.38 6.38 0 0 1-1.47-.61A4.2 4.2 0 0 1 4.66 14a7.8 7.8 0 0 1-.32-2.34c0-1.26.42-2.34 1.24-3.22a4.2 4.2 0 0 1 .1-3.19c.3-.1.76-.02 1.35.21A9.4 9.4 0 0 1 9 6.5a11.1 11.1 0 0 1 6 0l.6-.38c.4-.25.88-.48 1.43-.69.55-.2.98-.26 1.27-.17.47 1.18.5 2.24.12 3.19a4.56 4.56 0 0 1 1.24 3.22c0 .88-.11 1.67-.32 2.35a4.84 4.84 0 0 1-.83 1.64c-.34.41-.76.76-1.27 1.04-.5.28-1 .48-1.47.6-.47.13-1 .23-1.6.29.54.47.81 1.2.81 2.22v3.3c0 .18.07.34.2.46.13.13.34.17.62.11 2.41-.8 4.39-2.24 5.91-4.33A11.66 11.66 0 0 0 24 12.29c0-2.17-.54-4.18-1.61-6.02z\"/></svg>"
			},
			{
				"name": "dashboard",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 24 24\" fill=\"currentColor\"><path d=\"M13.33 0H24v8H13.33zM0 0h10.67v13.33H0zM0 16h10.67v8H0zM13.33 10.67H24V24H13.33z\"/></svg>"
			},
			{
				"name": "package",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 24 24\" fill=\"currentColor\"><path d=\"M13.97 12.19l-1.07.62V24l9.68-5.6V7.23zM2.05 5.75L5.78 7.9l9.95-5.74L12 0zM12 11.5l3.7-2.14 6.25-3.61-4.3-2.45-9.94 5.75.57.3zM5.17 9.4L1.5 7.28v11.1l9.61 5.56v-11.1l-4.15-2.4v2.72l-1.79-.92z\"/></svg>"
			},
			{
				"name": "function",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 24 24\" fill=\"currentColor\"><path d=\"M15.75 0S12.6 0 11.7 4.5l-.6 3H8.25L7.5 9h3.3L8.7 19.5c-.6 3-2.85 3-2.85 3h-1.5l-.6 1.5h4.5s3.15 0 4.05-4.5L14.4 9H18l.75-1.5H14.7l.6-3c.6-3 2.7-3 2.7-3h1.5l.75-1.5z\"/></svg>"
			},
			{
				"name": "codepen",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 24 24\" fill=\"currentColor\"><path d=\"M12.9.32a1.9 1.9 0 0 0-2.14.02L1.23 6.98C.72 7.34.4 7.92.4 8.54v7.16c0 .63.32 1.23.85 1.58l9.7 6.4a1.9 1.9 0 0 0 2.09 0l9.88-6.53c.54-.35.86-.94.86-1.58V8.54a1.9 1.9 0 0 0-.85-1.58zm7.38 7.75l-3.66 2.45-3.42-2.49-.03-4.9zM10.7 3.13v4.8l-3.52 2.46-3.5-2.35zM2.72 10.4L5 11.92l-2.27 1.59zm8.04 10.23l-7.08-4.75 3.54-2.46 3.54 2.37zM9.4 11.88l2.3-1.6 2.66 1.75L12 13.62zm3.84 8.75v-4.84l3.39-2.27 3.64 2.4zm8.04-7.04L18.89 12l2.4-1.61z\"/></svg>"
			},
			{
				"name": "vuetify",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 24 24\"><path fill=\"#1697f6\" d=\"M12 0H7.08L12 11.64\"/><path fill=\"#1867c0\" d=\"M12 0h4.93L12 11.64\"/><path fill=\"#7bc6ff\" d=\"M12 15v9L1.5 3.5H7\"/><path fill=\"#aeddff\" d=\"M12 15v9L22.5 3.5H17\"/></svg>"
			},
			{
				"name": "earth",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 24 24\" fill=\"currentColor\"><path d=\"M20.49 3.51A11.92 11.92 0 0 1 24 12c0 3.2-1.25 6.22-3.51 8.49A11.92 11.92 0 0 1 12 24c-3.2 0-6.22-1.25-8.49-3.51A11.92 11.92 0 0 1 0 12c0-3.2 1.25-6.22 3.51-8.49A11.92 11.92 0 0 1 12 0c3.2 0 6.22 1.25 8.49 3.51zm-1.07 15.91a10.43 10.43 0 0 0 2.87-5.32c-.32.47-.62.64-.8-.4-.2-1.7-1.76-.61-2.74-1.22-1.03.7-3.34-1.35-2.95.96.6 1.04 3.28-1.4 1.95.81-.85 1.54-3.11 4.94-2.82 6.71.04 2.57-2.63.54-3.54-.32-.62-1.7-.21-4.7-1.83-5.53-1.75-.08-3.26-.24-3.94-2.2-.4-1.4.44-3.49 1.94-3.8 2.2-1.39 2.99 1.61 5.05 1.67.64-.67 2.39-.89 2.53-1.64C13.8 8.9 16.86 8 15.01 7.5c-1.02.12-1.67 1.06-1.13 1.85-1.98.46-2.04-2.85-3.94-1.8-.05 1.65-3.1.53-1.05.2.7-.31-1.15-1.2-.15-1.04.49-.03 2.14-.6 1.7-1 .91-.57 1.68 1.37 2.58-.04.65-1.08-.27-1.28-1.08-.73-.46-.52.8-1.63 1.93-2.1.37-.17.73-.25 1-.23.56.65 1.6.76 1.66-.08a10.46 10.46 0 0 0-10.8 1.05c.49.22.76.5.3.85C5.67 5.5 4.2 6.95 2.9 6.75a10.38 10.38 0 0 0-1.3 3.76c1.08.36 1.33 1.06 1.1 1.3a3.55 3.55 0 0 0-1.07 1.9 10.42 10.42 0 0 0 2.94 5.71A10.43 10.43 0 0 0 12 22.5c2.8 0 5.44-1.1 7.42-3.08z\"/></svg>"
			}
		]
	}
}
