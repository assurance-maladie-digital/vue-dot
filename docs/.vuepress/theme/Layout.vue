<template>
	<div class="theme-container">
		<VApp>
			<VToolbar
				app
				dark
				fixed
				clipped-left
				color="secondary"
			>
				<VBtn
					icon
					class="mr-3"
					@click="sidebarShow = !sidebarShow"
				>
					<SvgIcon icon="menu" />
				</VBtn>

				<VBtn
					flat
					:to="$t('home')"
					active-class="active"
					class="px-3 ma-0 text-none"
				>
					<SvgIcon
						x-large
						icon="vue-dot-dark"
					/>
					<h1 class="ml-2 mr-2 font-weight-regular">{{ $t('name') }}</h1>
				</VBtn>

				<VSpacer />

				<VTextField
					solo
					light
					:label="$t('toolbar.search')"
					single-line
					hide-details
					class="search mr-3"
					append-icon="search"
				/>

				<VMenu
					left
					bottom
					offset-y
				>
					<VBtn
						flat
						slot="activator"
						class="px-3 ma-0 text-none"
					>
						{{ $t('toolbar.ecosystem') }}
						<SvgIcon
							size=".7em"
							icon="down-arrow"
							class="ml-2"
						/>
					</VBtn>

					<VList>
						<VListTile
							v-for="item in $t('toolbar.items')"
							:key="item.title"
							:href="item.href"
							target="_blank"
							rel="noopener noreferrer"
							ripple
						>
							<VListTileContent>
								<VListTileTitle>
									<SvgIcon
										v-if="item.icon"
										:icon="item.icon"
										class="mr-2"
										color="#888"
									/>

									<span>{{ item.title }}</span>
								</VListTileTitle>
							</VListTileContent>
						</VListTile>
					</VList>
				</VMenu>

				<VBtn
					flat
					class="px-3 ma-0 text-none"
				>
					v{{ version }}
				</VBtn>

				<LangBtn
					flat
					flags
					menu-left
					menu-bottom
					:outline="false"
					:value="$i18n.locale"
					@change="updateLang"
					:display-text-btn="false"
					:display-arrow="false"
					:available-languages="languages"
					class="ma-0 text-none lang-btn"
					flags-url="https://res.cloudinary.com/deraw/image/upload/v1547044454/"
				/>
			</VToolbar>

			<VNavigationDrawer
				app
				fixed
				clipped
				:value="sidebar"
				v-model="sidebarShow"
			>
				<VList expand>
					<template v-for="item in $t('toolbar.sidebarItems')">
						<VListTile
							v-if="!item.items"
							:key="item.title"
							:to="item.to || null"
							:href="item.href || null"
							:target="item.href ? '_blank' : null"
							:rel="item.href ? 'noopener noreferrer' : null"
							ripple
							exact
						>
							<VListTileContent>
								<VListTileTitle>
									<SvgIcon
										v-if="item.icon"
										:icon="item.icon"
										class="mr-3"
									/>

									<span>{{ item.title }}</span>
								</VListTileTitle>
							</VListTileContent>
						</VListTile>

						<VListGroup
							v-if="item.items"
							:group="item.group"
							no-action
						>
							<VListTile
								:key="item.title"
								:to="item.to"
								slot="activator"
								ripple
								exact
							>
								<VListTileContent>
									<VListTileTitle>
										<SvgIcon
											v-if="item.icon"
											:icon="item.icon"
											class="mr-3"
										/>

										<span>{{ item.title }}</span>
									</VListTileTitle>
								</VListTileContent>
							</VListTile>

							<VListTile
								v-for="item in item.items"
								:key="item.title"
								:to="item.to"
								ripple
								exact
							>
								<VListTileContent>
									<VListTileTitle>
										<span>{{ item.title }}</span>
									</VListTileTitle>
								</VListTileContent>
							</VListTile>
						</VListGroup>
					</template>
				</VList>
			</VNavigationDrawer>

			<VContent>
				<component :is="layout" />
			</VContent>

			<VFooter
				v-if="$page.frontmatter.home"
				app
				dark
				absolute
				height="auto"
				class="px-3 py-5"
			>
				<VLayout
					column
					align-center
					justify-center
					class="footer-content"
				>
					<img
						:src="$withBase('/img/logo-am.svg')"
						height="70"
					>

					<p class="mt-3 mb-0">
						{{ $page.frontmatter.footerRealease }}
						<a href="https://github.com/assurance-maladie-digital/vue-dot/blob/master/LICENSE">
							{{ $page.frontmatter.footerRealeaseLink }}
						</a>
					</p>
					<p class="mb-0">
						{{ $page.frontmatter.footerMadeWith }}
						<a href="https://fr.vuejs.org/">
							{{ $page.frontmatter.footerVueJs }}
						</a>
					</p>
					<p class="mb-0">{{ $page.frontmatter.footerCopyright }}</p>
				</VLayout>
			</VFooter>
		</VApp>
	</div>
</template>

<script>
	import Home from './layouts/Home.vue';
	import Basic from './layouts/Basic.vue';
	import { version } from '../../../package.json';
	import { loadLanguageAsync } from '../i18n';

	export default {
		name: 'Layout',
		components: {
			Home,
			Basic
		},
		data() {
			return {
				version,
				initiated: false,
				languages: ['en', 'fr'],
				sidebarShow: false
			};
		},
		computed: {
			layout() {
				if (this.$page.frontmatter.home) {
					return 'Home';
				}

				return this.$page.frontmatter.layout || 'Basic'
			},
			sidebar() {
				return this.sidebarShow;
			}
		},
		methods: {
			updateLang(lang) {
				loadLanguageAsync(lang);

				if (this.initiated) {
					this.$router.push(`/${lang}/`);
				} else {
					this.initiated = true;
				}
			},
			guard() {
				// If the route doesn't match selected langage, redirect
				if (!this.$route.path.includes(this.$i18n.locale)) {
					this.$router.push(`/${this.$i18n.locale}/`);
				}
			}
		},
		watch: {
			$page() {
				this.sidebarShow = !this.$page.frontmatter.home;
			}
		},
		created() {
			if (!this.$vuetify.breakpoint.xsOnly) {
				this.sidebarShow = !this.$page.frontmatter.home;
			}

			// Redirect default route
			if (this.$route.path === '/') {
				this.$router.push(`/${this.$i18n.locale}/`);
			}

			this.guard();
		}
	}
</script>

<style scoped>
	.v-toolbar {
		z-index: 5 !important;
	}

	.v-toolbar .v-btn:not(.v-btn--icon),
	.v-menu,
	.v-menu >>> .v-menu__activator {
		height: 100%;
	}

	.footer-content {
		opacity: .9;
	}

	.v-footer a {
		color: #bbb;
	}

	.v-list__tile__title {
		display: flex;
		align-items: center;
	}

	.v-navigation-drawer >>> .v-list .v-list__tile {
		transition: background .15s !important;
	}

	.v-input.search {
		flex: none;
		height: 38px;
		width: 250px;
		overflow: hidden;
		border-radius: 50px;
	}

	.v-input.search >>> .v-input__control {
		min-height: 38px !important;
	}

	.lang-btn >>> .v-btn {
		margin: 0;
		height: 100%;
		padding: 20px;
		min-width: 50px;
	}
</style>

<style lang="stylus">
	@require '~vuetify/src/stylus/main'
	@require '../../../dist/vue-dot.css'

	@import url('https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Material+Icons')
</style>
