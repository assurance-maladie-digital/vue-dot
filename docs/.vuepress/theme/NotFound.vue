<template>
	<VApp>
		<Toolbar />

		<VContent>
			<VContainer>
				 <VLayout align-center>
					<VFlex text-xs-center>
						<h1 class="display-2 primary--text mt-5 mb-3">{{ $t('error404.title') }}</h1>

						<p>{{ $t('error404.description') }}</p>

						<VBtn
							:to="`/${this.$i18n.locale}/guide/`"
							color="primary"
							outline
							exact
						>
							{{ $t('error404.button') }}
						</VBtn>
					</VFlex>
				</VLayout>
			</VContainer>
		</VContent>
	</VApp>
</template>

<script>
	import Toolbar from '../components/Toolbar.vue';

	export default {
		name: 'NotFound',
		components: {
			Toolbar
		},
		mounted() {
			// Redirect default route
			if (this.$route.path === '/') {
				this.$router.push(`/${this.$i18n.locale}/`);
			}
		}
	};
</script>
