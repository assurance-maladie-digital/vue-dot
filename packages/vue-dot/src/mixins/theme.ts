import Vue from 'vue';

const theme = {
	created(this: Vue) {
		if (this.$theme && this.$theme.config.colors) {
			this.$vuetify.theme = this.$theme.config.colors;
		}
	}
};

export default theme;
