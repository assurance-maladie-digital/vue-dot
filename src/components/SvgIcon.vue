<template>
	<span
		:class="{
			'large': large,
			'medium': medium,
			'small': small,
			'x-large': xLarge,
			'custom-color': color
		}"
		:style="{
			color,
			width: size,
			height: size
		}"
		class="x-svg-icon"
		aria-hidden="true"
	>
		<slot>
			<!-- Custom icons from theme, only render if it's looks like svg -->
			<template v-if="$theme">
				<div
					v-for="iconTheme in themeIcons"
					:key="iconTheme.name"
					v-html="iconTheme.svg"
				/>
			</template>
		</slot>
	</span>
</template>

<script lang="ts">
	import Vue from 'vue';

	interface Icon extends Object {
		name: string;
		svg: string;
	}

	export default Vue.extend({
		name: 'SvgIcon',
		props: {
			color: {
				type: String,
				default: 'currentColor'
			},
			icon: {
				type: String,
				default: undefined
			},
			large: {
				type: Boolean,
				default: false
			},
			medium: {
				type: Boolean,
				default: false
			},
			size: {
				type: String,
				default: ''
			},
			small: {
				type: Boolean,
				default: false
			},
			xLarge: {
				type: Boolean,
				default: false
			}
		},
		computed: {
			themeIcons(): object {
				// If there is icons in theme
				if (this.$theme && this.$theme.config && this.$theme.config.icons) {
					const filtered = this.$theme.config.icons.filter((icon: Icon) => {
						return this.icon === icon.name;
					});

					return filtered;
				}

				return {};
			}
		}
	});
</script>

<style lang="scss">
	$default: 1.35em;
	$small: 1em;
	$medium: 1.5em;
	$large: 1.85em;
	$xLarge: 2.25em;

	.x-svg-icon {
		width: $default;
		height: $default;
		position: relative;
		display: inline-block;
		vertical-align: middle;

		svg {
			left: 0;
			bottom: 0;
			width: 100%;
			height: 100%;
			position: absolute;
			transition: fill .25s;
		}

		&.custom-color svg {
			fill: currentColor;
		}

		&.small {
			width: $small;
			height: $small;
		}

		&.medium {
			width: $medium;
			height: $medium;
		}

		&.large {
			width: $large;
			height: $large;
		}

		&.x-large {
			width: $xLarge;
			height: $xLarge;
		}
	}
</style>
