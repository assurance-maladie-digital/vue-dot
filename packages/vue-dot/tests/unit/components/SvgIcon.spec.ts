import Vue from 'vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import SvgIcon from 'vue-dot-src/components/SvgIcon.vue';

import Vuetify from 'vuetify';
import VueDot from 'vue-dot-src/index';
import { default as theme } from '../theme.json';

const localVue = createLocalVue();
Vue.use(Vuetify);

localVue.use(VueDot, {
	theme
});

describe('SvgIcon.vue', () => {
	const build = () => {
		const wrapper = shallowMount(SvgIcon, {
			localVue,
			propsData: {
				icon: 'home'
			}
		});

		return wrapper;
	};

	it('renders correctly', () => {
		const wrapper = build();

		expect(wrapper.html()).toMatchSnapshot();
	});

	it('renders an icon from the theme correctly', () => {
		const wrapper = build();
		wrapper.setProps({
			icon: 'vuejs'
		});

		expect(wrapper.html()).toMatchSnapshot();
	});
});
