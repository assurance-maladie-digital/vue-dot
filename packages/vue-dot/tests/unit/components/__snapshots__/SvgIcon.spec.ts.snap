// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgIcon.vue renders an icon from the theme correctly 1`] = `<span aria-hidden="true" class="svg-icon custom-color"><svg viewBox="0 0 24 24"><!----></svg> <div><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 400 400"><path fill="#4dba87" d="M237.42 86.66L207.19 139l-30.22-52.35H76.3l130.9 226.69L338.07 86.66z"></path><path fill="#435466" d="M237.42 86.66L207.19 139l-30.22-52.35h-48.3l78.52 136 78.53-136z"></path></svg></div></span>`;

exports[`SvgIcon.vue renders correctly 1`] = `<span aria-hidden="true" class="svg-icon custom-color"><svg viewBox="0 0 24 24"><path d="M12 1.8L0 12.6h3.6v9.6h6V15h4.8v7.2h6v-9.6H24z"></path></svg> </span>`;
